datalad_next/utils/deprecate.py

```diff
@@ -0,0 +1,89 @@
+import inspect
+import warnings
+from functools import wraps
+
+from datalad_next.utils import get_wrapped_class
+
+
+def deprecated(msg, version, parameter=None, parameter_choice=None):
+    """Annotate functions, classes, or function parameters with standardized
+    deprecation warnings.
+
+    Parameters
+    ----------
+    version: str
+      Software version number at which the deprecation was made
+    msg: str
+      Custom message to append to a deprecation warning
```

Comment:
```suggestion
    msg: str
      Custom message to append to a deprecation warning
    version: str
      Software version number at which the deprecation was made
```

Reordered to match signature.